src/derivative_wrappers.jl

```diff
@@ -38,15 +38,9 @@ end
 
 jacobian_autodiff(f, x, odefun) = (ForwardDiff.derivative(f,x),1)
 function jacobian_autodiff(f, x::AbstractArray, odefun)
-  if DiffEqBase.has_colorvec(odefun)
-    colorvec = odefun.colorvec
-    sparsity = odefun.jac_prototype
-    jac_prototype = nothing
-  else
-    colorvec = 1:length(x)
-    sparsity = nothing
-    jac_prototype = odefun.jac_prototype
-  end
+  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec : 1:length(x)
+  sparsity = odefun.jac_prototype
+  jac_prototype = odefun.jac_prototype
```

Comment:
Sadly we chose to use `jac_prototype` to represent both sparsity and the real *jac_prototype*, so we got a dilemma that someone may have a sparseCSC for sparsity but still wish to keep a dense jacobian, while others want both sparsity and jacobian to be sparse matrix.

Reply:
maybe in the DiffEqFunction we should just default to sparsity=jac_prototype, but allow it to be changed independently?

Reply:
Makes sense, just like those in oop colored jacobian.

Reply:
it would definitely not be the most used combination, but there are times when I can see it viable (I wanted it last week, so that means it does show up!)

Reply:
Thanks to @huanglangwen, this should now be resolved.